cdc/sink/mq.go

```diff
@@ -297,16 +322,15 @@ func (k *mqSink) run(ctx context.Context) error {
 	return wg.Wait()
 }
 
-func (k *mqSink) writeToProducer(ctx context.Context, message *codec.MQMessage, partition int32) error {
-	if partition >= 0 {
-		err := k.mqProducer.AsyncSendMessage(ctx, message, partition)
-		if err != nil {
-			return err
-		}
-		return k.mqProducer.Flush(ctx)
+// asyncWriteToProducer writes message to kafka producer asynchronously.
+func (k *mqSink) asyncWriteToProducer(
+	ctx context.Context, topic string, partition int32, message *codec.MQMessage,
+) error {
+	err := k.mqProducer.AsyncSendMessage(ctx, topic, partition, message)
+	if err != nil {
+		return err
 	}
-
-	return k.mqProducer.SyncBroadcastMessage(ctx, message)
+	return k.mqProducer.Flush(ctx)
```

Comment:
Is this a synchronous send, since it calls `Flush` at last.
Besides the function is only used  in L299, the parameter `partition` can be removed.

Reply:
Fixed.